presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -1175,12 +1174,6 @@ public Object visitCast(Cast node, Object context)
                 return value;
             }
 
-            // hack!!! don't optimize CASTs for types that cannot be represented in the SQL AST
-            // TODO: this will not be an issue when we migrate to RowExpression tree for this, which allows arbitrary literals.
-            if (optimize && !isSupportedLiteralType(type(node))) {
```

Comment:
This is not as easy per https://github.com/prestosql/presto/pull/5673, but for now it should be enough to add long timestamp tz's class to supported types.